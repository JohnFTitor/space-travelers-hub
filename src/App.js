import './App.css';
import NavBar from './components/NavBar';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <NavBar />
      </header>
    </div>
  );
}

export default App;
