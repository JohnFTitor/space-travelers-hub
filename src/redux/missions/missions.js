const ADD_MISSIONS = 'spaceTravelersHub/missions/ADD_MISSION';
const COMPLETE_STATUS = 'spaceTravelersHub/missions/COMPLETE_STATUS';

const initialState = {
  data: [],
  status: 'iddle',
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ADD_MISSIONS:
      return { ...state, data: [...state.data, ...action.payload] };
    case COMPLETE_STATUS:
      return { ...state, status: 'completed' };
    default:
      return state;
  }
};

export default reducer;

const addMissions = (missionsArray) => ({
  type: ADD_MISSIONS,
  payload: missionsArray,
});

const completeStatus = () => ({
  type: COMPLETE_STATUS,
});

export { addMissions, completeStatus };

const getMissions = () => (async (dispatch) => {
  const response = await fetch('https://api.spacexdata.com/v3/missions');
  const data = await response.json();
  const missionsArray = data.map((mission) => ({
    id: mission.mission_id,
    name: mission.mission_name,
    description: mission.description,
  }));
  dispatch(addMissions(missionsArray));
  dispatch(completeStatus());
});

export { getMissions };
