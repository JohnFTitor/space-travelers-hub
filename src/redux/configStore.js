import { createStore, applyMiddleware, combineReducers } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import missions from './missions/missions';

const reducer = combineReducers({
  missions,
});

const store = createStore(reducer, applyMiddleware(thunk, logger));

export default store;
