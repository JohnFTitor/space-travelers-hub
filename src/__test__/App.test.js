import React from 'react';
import {
  waitFor,
  render,
  fireEvent,
  screen,
  cleanup,
} from '@testing-library/react';
import App from '../App';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { createStore, applyMiddleware, combineReducers } from 'redux';
import thunk from 'redux-thunk';
import missions from '../redux/missions/missions';
import rocketsReducer from '../redux/rockets/rockets';
import '@testing-library/jest-dom';

const reducer = combineReducers({
  missions,
  rocketsReducer,
});

const store = createStore(reducer, applyMiddleware(thunk));

afterEach(() => cleanup());

describe('Rockets and profile rockets', () => {
  test('Rockets are rendering', async () => {
    const { container } = render(
      <BrowserRouter>
        <Provider store={store}>
          <App />
        </Provider>
      </BrowserRouter>
    );

    await waitFor(() => screen.getByText('Falcon 1'));

    expect(container).toMatchSnapshot();
  });

  it('Profile renders reservations', async () => {
    render(
      <BrowserRouter>
        <Provider store={store}>
          <App />
        </Provider>
      </BrowserRouter>
    );

    fireEvent.click(screen.getByText('Rockets'));

    await waitFor(() => screen.getByText('Falcon 1'));

    fireEvent.click(screen.getByTestId(1));

    fireEvent.click(screen.getByText('My profile'));

    expect(screen.getByText('Falcon 1')).toBeInTheDocument();
  });
});

it('NavBar changes the section', async () => {
  render(
    <BrowserRouter>
      <Provider store={store}>
        <App />
      </Provider>
    </BrowserRouter>
  );

  fireEvent.click(screen.getByText('Missions'));

  await waitFor(() => screen.getByText('Thaicom'));

  expect(screen.getByTestId('missions-section')).toMatchSnapshot();
});
