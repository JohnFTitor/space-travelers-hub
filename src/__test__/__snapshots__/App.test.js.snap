// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar changes the section 1`] = `
<section
  class="missions"
  data-testid="missions-section"
>
  <table>
    <thead>
      <tr>
        <th>
           Mission 
        </th>
        <th>
           Description 
        </th>
        <th>
           Status 
        </th>
        <th>
           
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th>
          Thaicom
        </th>
        <th>
          <p>
            Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology.
          </p>
          <br />
        </th>
        <th
          class="badge"
        >
          <span>
            not a member
          </span>
        </th>
        <th
          class="join"
        >
          <button
            id="9D1B7E0"
            type="button"
          >
            Join Mission
          </button>
        </th>
      </tr>
      <tr>
        <th>
          Telstar
        </th>
        <th>
          <p>
            Telstar 19V (Telstar 19 Vantage) is a communication satellite in the Telstar series of the Canadian satellite communications company Telesat. It was built by Space Systems Loral (MAXAR) and is based on the SSL-1300 bus. As of 26 July 2018, Telstar 19V is the heaviest commercial communications satellite ever launched, weighing at 7,076 kg (15,600 lbs) and surpassing the previous record, set by TerreStar-1 (6,910 kg/15230lbs), launched by Ariane 5ECA on 1 July 2009.
          </p>
          <br />
        </th>
        <th
          class="badge"
        >
          <span>
            not a member
          </span>
        </th>
        <th
          class="join"
        >
          <button
            id="F4F83DE"
            type="button"
          >
            Join Mission
          </button>
        </th>
      </tr>
      <tr>
        <th>
          Iridium NEXT
        </th>
        <th>
          <p>
            In 2017, Iridium began launching Iridium NEXT, a second-generation worldwide network of telecommunications satellites, consisting of 66 active satellites, with another nine in-orbit spares and six on-ground spares. These satellites will incorporate features such as data transmission that were not emphasized in the original design. The constellation will provide L-band data speeds of up to 128 kbit/s to mobile terminals, up to 1.5 Mbit/s to Iridium Pilot marine terminals, and high-speed Ka-band service of up to 8 Mbit/s to fixed/transportable terminals. The next-generation terminals and service are expected to be commercially available by the end of 2018. However, Iridium's proposed use of its next-generation satellites has raised concerns the service will harmfully interfere with GPS devices. The satellites will incorporate a secondary payload for Aireon, a space-qualified ADS-B data receiver. This is for use by air traffic control and, via FlightAware, for use by airlines. A tertiary payload on 58 satellites is a marine AIS ship-tracker receiver, for Canadian company exactEarth Ltd. Iridium can also be used to provide a data link to other satellites in space, enabling command and control of other space assets regardless of the position of ground stations and gateways.
          </p>
          <br />
        </th>
        <th
          class="badge"
        >
          <span>
            not a member
          </span>
        </th>
        <th
          class="join"
        >
          <button
            id="F3364BF"
            type="button"
          >
            Join Mission
          </button>
        </th>
      </tr>
      <tr>
        <th>
          Commercial Resupply Services
        </th>
        <th>
          <p>
            Commercial Resupply Services (CRS) are a series of contracts awarded by NASA from 2008–2016 for delivery of cargo and supplies to the International Space Station (ISS) on commercially operated spacecraft. The first CRS contracts were signed in 2008 and awarded $1.6 billion to SpaceX for 12 cargo transport missions and $1.9 billion to Orbital Sciences for 8 missions, covering deliveries to 2016. In 2015, NASA extended the Phase 1 contracts by ordering an additional three resupply flights from SpaceX and one from Orbital Sciences. After additional extensions late in 2015, SpaceX is currently scheduled to have a total of 20 missions and Orbital 10. SpaceX began flying resupply missions in 2012, using Dragon cargo spacecraft launched on Falcon 9 rockets from Space Launch Complex 40 at Cape Canaveral Air Force Station, Cape Canaveral, Florida. Orbital Sciences began deliveries in 2013 using Cygnus spacecraft launched on the Antares rocket from Launch Pad 0A at the Mid-Atlantic Regional Spaceport (MARS), Wallops Island, Virginia. A second phase of contracts (known as CRS2) were solicited and proposed in 2014. They were awarded in January 2016 to Orbital ATK, Sierra Nevada Corporation, and SpaceX, for cargo transport flights beginning in 2019 and expected to last through 2024.
          </p>
          <br />
        </th>
        <th
          class="badge"
        >
          <span>
            not a member
          </span>
        </th>
        <th
          class="join"
        >
          <button
            id="EE86F74"
            type="button"
          >
            Join Mission
          </button>
        </th>
      </tr>
      <tr>
        <th>
          SES
        </th>
        <th>
          <p>
            SES S.A. is a communications satellite owner and operator providing video and data connectivity worldwide to broadcasters, content and internet service providers, mobile and fixed network operators, governments and institutions, with a mission to “connect, enable, and enrich”. SES operates more than 50 geostationary orbit satellites and 16 medium Earth orbit satellites. These comprise the well-known European Astra TV satellites, the O3b data satellites and others with names including AMC, Ciel, NSS, Quetzsat, YahSat and SES.
          </p>
          <br />
        </th>
        <th
          class="badge"
        >
          <span>
            not a member
          </span>
        </th>
        <th
          class="join"
        >
          <button
            id="6C42550"
            type="button"
          >
            Join Mission
          </button>
        </th>
      </tr>
      <tr>
        <th>
          JCSAT
        </th>
        <th>
          <p>
            The JSAT constellation is a communication and broadcasting satellite constellation formerly operated by JSAT Corporation and currently by SKY Perfect JSAT Group. It has become the most important commercial constellation in Japan, and the fifth of the world. It has practically amalgamated all private satellite operators in Japan, with only B-SAT left as a local competitor.
          </p>
          <br />
        </th>
        <th
          class="badge"
        >
          <span>
            not a member
          </span>
        </th>
        <th
          class="join"
        >
          <button
            id="FE3533D"
            type="button"
          >
            Join Mission
          </button>
        </th>
      </tr>
      <tr>
        <th>
          AsiaSat
        </th>
        <th>
          <p>
            Asia Satellite Telecommunications Holdings Limited known as its brand name AsiaSat is a commercial operator of communication spacecraft. AsiaSat is based in Hong Kong but incorporated in Bermuda.
          </p>
          <br />
        </th>
        <th
          class="badge"
        >
          <span>
            not a member
          </span>
        </th>
        <th
          class="join"
        >
          <button
            id="593B499"
            type="button"
          >
            Join Mission
          </button>
        </th>
      </tr>
      <tr>
        <th>
          Orbcomm OG2
        </th>
        <th>
          <p>
            Orbcomm Generation 2 (OG2) second-generation satellites are intended to supplement and eventually replace the current first generation constellation. Eighteen satellites were ordered by 2008—nominally intended to be launched in three groups of six during 2010–2014—and by 2015 have all been launched, on three flights. Orbcomm has options for a further thirty OG2 spacecraft. The satellites were launched by SpaceX on the Falcon 9 launch system. Originally, they were to launch on the smaller Falcon 1e rocket.
          </p>
          <br />
        </th>
        <th
          class="badge"
        >
          <span>
            not a member
          </span>
        </th>
        <th
          class="join"
        >
          <button
            id="CE91D46"
            type="button"
          >
            Join Mission
          </button>
        </th>
      </tr>
      <tr>
        <th>
          ABS
        </th>
        <th>
          <p>
            ABS, formerly Asia Broadcast Satellite, is a global satellite operator based in Hong Kong and officially incorporated in Bermuda. Its services include direct-to-home and satellite-to-cable TV distribution, cellular services, and internet services. Operating 6 communication satellites, the satellite fleet currently covers 93% of the world’s population including the Americas, Africa, Asia Pacific, Europe, the Middle East, Russia and Commonwealth of Independent States.
          </p>
          <br />
        </th>
        <th
          class="badge"
        >
          <span>
            not a member
          </span>
        </th>
        <th
          class="join"
        >
          <button
            id="2CF444A"
            type="button"
          >
            Join Mission
          </button>
        </th>
      </tr>
      <tr>
        <th>
          Eutelsat
        </th>
        <th>
          <p>
            Eutelsat S.A. is a European satellite operator. Providing coverage over the entire European continent, the Middle East, Africa, Asia and the Americas, it is the world's third largest satellite operator in terms of revenues.
          </p>
          <br />
        </th>
        <th
          class="badge"
        >
          <span>
            not a member
          </span>
        </th>
        <th
          class="join"
        >
          <button
            id="F7709F2"
            type="button"
          >
            Join Mission
          </button>
        </th>
      </tr>
    </tbody>
  </table>
</section>
`;

exports[`Rockets and profile rockets Rockets are rendering 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <nav
        class="nav nav-parent"
      >
        <div
          class="logo-container"
        >
          <img
            alt="planet"
            class="planet"
            src="planet.png"
          />
          <h1>
            Space travelers' hub
          </h1>
        </div>
        <ul
          class="nav nav-child"
        >
          <li>
            <a
              aria-current="page"
              class="nav-link active"
              href="/"
              style="text-decoration: underline;"
            >
              Rockets
            </a>
          </li>
          <li>
            <a
              class="nav-link"
              href="/missions"
            >
              Missions
            </a>
          </li>
          <li>
            <a
              class="nav-link"
              href="/profile"
              id="profile"
            >
              My profile
            </a>
          </li>
        </ul>
      </nav>
    </header>
    <main>
      <section>
        <ul>
          <li
            class="rocket-li"
          >
            <div
              class="img-container"
            >
              <img
                alt="Falcon 1"
                src="https://imgur.com/DaCfMsj.jpg"
              />
            </div>
            <div
              class="rocket-info"
            >
              <h2>
                Falcon 1
              </h2>
              <div
                class="desc-txt"
              >
                <p>
                  The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.
                </p>
              </div>
              <button
                class="reserveBtn"
                data-testid="1"
                id="1"
                type="button"
              >
                reserve rocket
              </button>
            </div>
          </li>
          <li
            class="rocket-li"
          >
            <div
              class="img-container"
            >
              <img
                alt="Falcon 9"
                src="https://farm1.staticflickr.com/929/28787338307_3453a11a77_b.jpg"
              />
            </div>
            <div
              class="rocket-info"
            >
              <h2>
                Falcon 9
              </h2>
              <div
                class="desc-txt"
              >
                <p>
                  Falcon 9 is a two-stage rocket designed and manufactured by SpaceX for the reliable and safe transport of satellites and the Dragon spacecraft into orbit.
                </p>
              </div>
              <button
                class="reserveBtn"
                data-testid="2"
                id="2"
                type="button"
              >
                reserve rocket
              </button>
            </div>
          </li>
          <li
            class="rocket-li"
          >
            <div
              class="img-container"
            >
              <img
                alt="Falcon Heavy"
                src="https://farm5.staticflickr.com/4599/38583829295_581f34dd84_b.jpg"
              />
            </div>
            <div
              class="rocket-info"
            >
              <h2>
                Falcon Heavy
              </h2>
              <div
                class="desc-txt"
              >
                <p>
                  With the ability to lift into orbit over 54 metric tons (119,000 lb)--a mass equivalent to a 737 jetliner loaded with passengers, crew, luggage and fuel--Falcon Heavy can lift more than twice the payload of the next closest operational vehicle, the Delta IV Heavy, at one-third the cost.
                </p>
              </div>
              <button
                class="reserveBtn"
                data-testid="3"
                id="3"
                type="button"
              >
                reserve rocket
              </button>
            </div>
          </li>
          <li
            class="rocket-li"
          >
            <div
              class="img-container"
            >
              <img
                alt="Starship"
                src="https://live.staticflickr.com/65535/48954138962_ee541e6755_b.jpg"
              />
            </div>
            <div
              class="rocket-info"
            >
              <h2>
                Starship
              </h2>
              <div
                class="desc-txt"
              >
                <p>
                  Starship and Super Heavy Rocket represent a fully reusable transportation system designed to service all Earth orbit needs as well as the Moon and Mars. This two-stage vehicle — composed of the Super Heavy rocket (booster) and Starship (ship) — will eventually replace Falcon 9, Falcon Heavy and Dragon.
                </p>
              </div>
              <button
                class="reserveBtn"
                data-testid="4"
                id="4"
                type="button"
              >
                reserve rocket
              </button>
            </div>
          </li>
        </ul>
      </section>
    </main>
  </div>
</div>
`;
